doc/design/design_submit_job.md

```diff
@@ -0,0 +1,60 @@
+# Submit SQLFLow SQL Asynchronously
+
+## Abstract
+
+In the current system, SQLFlow client connects the SQLFlow server with long links. the SQLFlow client sends
+a request which contains the user-typed SQL and hung until the SQLFlow server finishes executing the SQL.
+Sometimes, it would cost a long time to execute the SQL and lead to gRPC time out.
+The purpose of this proposal is to introduce a way to submit the SQLFlow SQL asynchronously to avoid the
+time out issue.
+
+## High-Level Design
+
+For each request, the SQLFlow code generator would generate a submitter program in Python, which submit a Job to the Spark/k8s cluster or launch a process on the host. For the cluster job, submitter program would return an URL which is used to monitor the task status, and the SQLflow server would response the URL to SQLFlow client, and
```

Comment:
What is the "each request"? Is it for a SQL statement, SQL statements in a Jupyter Notebook cell, or all SQL statements in a notebook?